envelope-encryption-sample/python-cli/cli.py

```diff
@@ -0,0 +1,393 @@
+            "name": kek_name,
+            "plaintext": plaintext_bytes,
+            "plaintext_crc32c": plaintext_crc32c,
+        }
+    )
+
+    # Optional, but recommended:
+    # perform integrity verification on encrypt_response.
+    # For more details on ensuring E2E in-transit
+    # integrity to and from Cloud KMS visit:
+    # https://cloud.google.com/kms/docs/data-integrity-guidelines
+    if not encrypt_response.verified_plaintext_crc32c:
+        raise Exception(
+            "The request sent to the server was corrupted in-transit."
+        )
+    if not encrypt_response.ciphertext_crc32c == crc32c(
+        encrypt_response.ciphertext
+    ):
+        raise Exception(
+            "The response received from the server was corrupted in-transit."
+        )
+    # End integrity verification
+
+    return encrypt_response
+
+
+def gcp_decrypt_symmetric(
+    project_id: str,
+    location: str,
+    keyring_name: str,
+    kek_name: str,
+    ciphertext: bytes,
+) -> kms.DecryptResponse:
+    """
+    Decrypt the ciphertext using the symmetric key stored in GCP KMS
+
+    Args:
+        project_id (string): Google Cloud project ID.
+        location (string): Cloud KMS location.
+        keyring_name (string): Name of the Cloud KMS key ring.
+        kek_name (string): Name of the key to use.
+        ciphertext (bytes): Encrypted bytes to decrypt.
+
+    Returns:
+        DecryptResponse: Response including plaintext.
+
+    """
+
+    # Create the client.
+    client = kms.KeyManagementServiceClient()
+
+    # Build the key name.
+    kek_name = client.crypto_key_path(
+        project_id, location, keyring_name, kek_name
+    )
+
+    # Optional, but recommended: compute ciphertext's CRC32C.
+    # See crc32c() function defined below.
+    ciphertext_crc32c = crc32c(ciphertext)
+
+    # Call the API.
+    decrypt_response = client.decrypt(
+        request={
+            "name": kek_name,
+            "ciphertext": ciphertext,
+            "ciphertext_crc32c": ciphertext_crc32c,
+        }
+    )
+
+    # Optional, but recommended:
+    # perform integrity verification on decrypt_response.
+    # For more details on ensuring E2E in-transit integrity
+    # to and from Cloud KMS visit:
+    # https://cloud.google.com/kms/docs/data-integrity-guidelines
+    if not decrypt_response.plaintext_crc32c == crc32c(
+        decrypt_response.plaintext
+    ):
+        raise Exception(
+            "The response received from the server was corrupted in-transit."
+        )
+    # End integrity verification
+
+    return decrypt_response
+
+
+def local_encrypt_symmetric(
+    data_encryption_key: bytes, plaintext: str
+) -> bytes:
+    """
+    Encrypt plaintext using a symmetric key.
+
+    Args:
+        data_encryption_key (bytes): DEK bytes to be used on encrypt process.
+        plaintext (string): message to encrypt
+
+    Returns:
+        dict: base64 encrypted ciphertext.
+
+    """
+    f = Fernet(data_encryption_key)
```

Comment:
Let's add a note here and in one of the READMEs (maybe the main one) that "the local encryption/decryption is done using Pyca's Fernet package, which uses AES128-CBC + HMAC-SHA256 behind the scenes". Good for the customer to know, I had to look up what it was using out of curiosity.